test/check_cpp/CMakeLists.txt

```diff
@@ -0,0 +1,15 @@
+if(NOT BUILD_TESTING)
+  return()
+endif()
+
```

Comment:
There are failures on Travis for clang with sanitize. Not exactly sure why. So maybe try to build statically, so add [this](https://github.com/nats-io/nats.c/blob/a9aef130ec008dc4f8fed16b4f7386000b71b9b8/test/CMakeLists.txt#L9-L17) to this make file.